### PR TITLE
feat: implement skybox texture

### PR DESCRIPTION
**Pull request by dziamidchyk**
Implement skybox texture

Example:

### CODE REVIEW

File: package/cpp/core/RNFEngineImpl.Skybox.cpp [hunk 1]
```diff
@@ -29,9 +33,41 @@ void EngineImpl::createAndSetSkybox(std::string hexColor, std::optional<bool> sh
...
+
+  builder.environment(cubemap);
+
+  Skybox* skybox = builder.build(*_engine);
+  _skybox = References<Skybox>::adoptEngineRef(_engine, skybox, [](std::shared_ptr<Engine> engine, Skybox* skybox) {
+    Logger::log(TAG, "Destroying Skybox...");
+    engine->destroy(skybox);
+  });
+  _scene->setSkybox(_skybox.get());
 }
```
Comment (hannojg) [inline package/cpp/core/RNFEngineImpl.Skybox.cpp:71]: We actually also need to destroy / deallocate the texture: 

```suggestion
  Skybox* skybox = builder.build(*_engine);
  _skybox = References<Skybox>::adoptEngineRef(_engine, skybox, [cubemap](std::shared_ptr<Engine> engine, Skybox* skybox) {
    Logger::log(TAG, "Destroying Skybox...");
    engine->destroy(skybox);
    engine->destroy(cubemap);
  });
  _scene->setSkybox(_skybox.get());
}
```

Note: I think we have some other places in the code base where we miss to do that 😅

**Review by hannojg — CHANGES_REQUESTED** [review]
Thank you so much for the PR! 

Just one little change needed, then we are good to go!

File: package/cpp/core/RNFEngineImpl.Skybox.cpp (hunk 1 above)
Comment (dziamidchyk) [reply package/cpp/core/RNFEngineImpl.Skybox.cpp]: Yeah makes sense, updated the PR :+1:

**Review by hannojg — APPROVED** [review]

**Comment by hannojg** [comment]
Awesome, will be part of the next release, thanks!

**Pull request merged**